pyo3-macros-backend/src/pyclass.rs

```diff
@@ -471,13 +515,74 @@ fn impl_enum_class(
             fn __pyo3__repr__(&self) -> &'static str {
                 match self {
                     #(#variants_repr)*
-                    _ => unreachable!("Unsupported variant type."),
+                    _ => ::core::unreachable!("Unsupported variant type."),
```

Comment:
Can you give me an example where this match arm is necessary? Seems like something we shouldn't need to have in the codegen.

```suggestion
```

Reply:
As an addition, I suspect that this is what has caused the MSRV failure; this macro had some hygiene fixes in 1.50: https://github.com/rust-lang/rust/pull/78343

Reply:
It was for #417. Will remove that for this PR.